web/reducers/environment.js

```diff
@@ -42,6 +45,21 @@ export default function reducer(state = initialState, action) {
             ...state,
             fetchingData: action.fetchingData
         }
+    case EDIT_MODE_SCHEDULE:
+        return {
+            ...state,
+            editModeSchedule: action.editModeSchedule
+        }
+    case UPDATE_BLOCK_FRACTION:
+        return {
+            ...state,
+            currentTeamMember: {
+                ...state.currentTeamMember,
+                weeklySetblocks: state.currentTeamMember.weeklySetblocks.map(
```

Comment:
I haven't usually seen state changes that go this deep.. Should be okay if nothing is breaking, but to minimize the complexity in the reducers, you may want to consider breaking out the state structures like so:

Instead of just
```
const initialState = {
    currentTeamMember: {
        id: '',
        name: '',
        weeklySetblocks: []
    },
    ...
}
```
you could keep a redundant shallow version of weekly setblocks
```
const initialState = {
    currentTeamMember: {
        id: '',
        name: ''
    },
    currentWeeklySetblocks: []
    ...
}
```
This way your UPDATE_BLOCK_FRACTION doesn't have to go the extra layer deep.

This is just a suggestion, not critical

Reply:
Yes, good approach, the reducer will be more easy to read